openfoodfoundation/openfoodnetwork
# Consider deleted products when creating permalinks

#### What? Why?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/3844



Spree's implementation and our implementation to create a unique permalink failed to notice conflicts with soft-deleted products. This patch looks at deleted products as well.



#### What should we test?


Test product import:
- Upload products with Rachel's test sheet: https://docs.google.com/spreadsheets/d/1-n5PyoOqloRXWdw-DUGMoy0AWs5KeeQWKITnMIbDyyg/edit#gid=788111594
- Delete some of the products.
- Upload the products again.
- It should succeed.
- The new products have a different permalink to the products that got deleted.

Test enterprise groups:
- Try to assign the same permalink to two groups.
- A new permalink should be created and the data should save.

#### Release notes


Fixed: Creating new products could fail if a product with the same name was deleted before.



Changelog Category: Fixed
